torch/nn/functional.py

```diff
@@ -3916,6 +3916,12 @@ def interpolate(input: Tensor, size: Optional[int] = None, scale_factor: Optiona
     if antialias and not (mode in ("bilinear", "bicubic") and input.ndim == 4):
         raise ValueError("Anti-alias option is only supported for bilinear and bicubic modes")
 
+    # Logging logic
+    is_channels_last = input.is_contiguous(memory_format=torch.channels_last)
+    is_image_or_mask = input.ndim == 4 and input.shape[1] < 4
+    log_string = f"torch.nn.functional.interpolate_dtype={input.dtype}_mode={mode}_antialias={antialias}_channelslast={is_channels_last}_imageormask={is_image_or_mask}"
```

Comment:
I don't think this is ok. Creating such string is quite expensive and I don't think we want all our users to pay that price.

Reply:
I understand this may not be desirable on more ubiquitous ops like `sum()` or `max()`, but is this a problem in practice for `interpolate()` which is in general a fairly slow operation? On a typical input the string construction would only account for 0.1% of the `interpolate()` call:

```py
[ins] In [13]: import torch

[ins] In [14]: input = torch.rand(1, 3, 350, 280); antialias = True; mode = "bilinear"

[ins] In [16]: torch.set_num_threads(1)

[ins] In [17]: %timeit torch.nn.functional.interpolate(input, size=(224, 224), mode=mode, antialias=antialias)
1.05 ms ± 449 ns per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

[ins] In [18]: %%timeit
          ...:     is_channels_last = input.is_contiguous(memory_format=torch.channels_last)
          ...:     is_image_or_mask = input.ndim == 4 and input.shape[1] < 4
          ...:     log_string = f"torch.nn.functional.interpolate_dtype={input.dtype}_mode={mode}_antialias={antialias}
          ...: _channelslast={is_channels_last}_imageormask={is_image_or_mask}"
          ...: 
1.15 µs ± 1.97 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```

If you feel strongly against this, would you mind sharing your thoughts on [this thread](https://fb.workplace.com/groups/1144215345733672/posts/2382133531941841/?comment_id=2382227655265762&reply_comment_id=2382864245202103)? According to our internal discussion with @jisaacso  and @colin2328 , this kind of logging should be possible?

Reply:
Sent there.